storage/block/result.go

```diff
@@ -161,6 +161,7 @@ func (s *fetchBlocksMetadataResults) Reset() {
 func (s *fetchBlocksMetadataResults) Close() {
 	for i := range s.results {
 		s.results[i].Blocks.Close()
+		s.results[i].ID.Finalize()
```

Comment:
I don't think the result `ID`s are cloned, we probably should, though, otherwise we might return the series ID to the pool